osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/alarm/ClearAlarmRegisterCommandExecutor.java

```diff
@@ -71,40 +69,55 @@ public AccessResultCode execute(
       final MessageMetadata messageMetadata)
       throws ProtocolAdapterException {
 
-    LOGGER.info(
-        "Clear alarm register by request for class id: {}, obis code: {}, attribute id: {}",
-        CLASS_ID,
-        OBIS_CODE,
-        ATTRIBUTE_ID);
+    final AttributeAddress alarmRegister1AttributeAddress =
+        this.dlmsObjectConfigService.getAttributeAddress(
+            device, DlmsObjectType.ALARM_REGISTER_1, null);
+
+    final AccessResultCode resultCodeAlarmRegister1 =
+        this.executeForAlarmRegister(conn, alarmRegister1AttributeAddress);
+
+    if (resultCodeAlarmRegister1 == null) {
+      throw new ProtocolAdapterException("Error occurred for clear alarm register 1.");
+    }
+
+    final Optional<AttributeAddress> optAlarmRegister2AttributeAddress =
+        this.dlmsObjectConfigService.findAttributeAddress(
+            device, DlmsObjectType.ALARM_REGISTER_2, null);
+
+    if (!optAlarmRegister2AttributeAddress.isPresent()) {
+      return resultCodeAlarmRegister1;
+    } else {
+      final AccessResultCode resultCodeAlarmRegister2 =
+          this.executeForAlarmRegister(conn, optAlarmRegister2AttributeAddress.get());
+
+      if (resultCodeAlarmRegister2 != null) {
+        return resultCodeAlarmRegister2;
```

Comment:
Only the result of reading AlarmRegister2 is returned. Shouldn't the result of reading AlarmRegister1 also be returned?

Reply:
Good onbe, If AlarmRegister 1 is returned without success than this will be returned now